packages/resource-deleter/test/main.spec.js

```diff
@@ -0,0 +1,166 @@
+import silentLogger from '../src/utils/silent-logger'
+import ResourceDeleter from '../src/main'
+
+describe('::ResourceDeleter', () => {
+  const logger = {
+    ...silentLogger,
+  }
+
+  let resourceDeleter
+  beforeEach(() => {
+    const options = {
+      apiConfig: {
+        projectKey: 'sample-test-project',
+      },
+      resource: 'carts',
+      logger,
+    }
+    resourceDeleter = new ResourceDeleter(options)
+  })
+
+  describe('::constructor', () => {
+    test('should be a function', () => {
+      expect(typeof ResourceDeleter).toBe('function')
+    })
+    test('should set default properties', () => {
+      expect(resourceDeleter.apiConfig).toEqual({
+        projectKey: 'sample-test-project',
+      })
+      expect(resourceDeleter.logger).toEqual(logger)
+    })
+    test('should throw error if no `apiConfig` in `options` parameter', () => {
+      expect(() => new ResourceDeleter({})).toThrow(
+        /The constructor must passed an `apiConfig` object/
+      )
+    })
+
+    test('should throw error if no `resource` in `options` parameter', () => {
+      expect(
+        () =>
+          new ResourceDeleter({
+            apiConfig: {
+              projectKey: 'sample-test-project',
+            },
+            logger,
+          })
+      ).toThrow(/A `resource` object must be passed/)
+    })
+  })
+
+  describe('::run', () => {
+    let payload
+    describe('with status code 200', () => {
+      beforeEach(() => {
+        payload = {
+          statusCode: 200,
+          body: {
+            results: [
+              { id: 'foo1', key: 'fooKey', version: 1 },
+              { id: 'boo2', key: 'booKey', version: 2 },
+              { id: 'fooboo3', key: 'foboKey', version: 3 },
+            ],
+          },
+        }
+
+        resourceDeleter.client.execute = jest.fn(() => Promise.resolve(payload))
+      })
+
+      test('should delete fetched resource', async () => {
+        const data = await resourceDeleter.run('sample-test-project')
+        expect(data).toBeDefined()
+        expect(data).toBeTruthy()
+        expect(data).toHaveLength(0)
+      })
+    })
+
+    describe('should throw error when resource is empty with status code 200', () => {
+      beforeEach(() => {
+        payload = {
+          statusCode: 200,
+          body: {
+            results: [],
+          },
+        }
+
+        resourceDeleter.client.execute = jest.fn(() => Promise.resolve(payload))
+      })
+
+      test('should throw error for empty resource', () => {
+        expect(resourceDeleter.run('sample-test-project')).rejects.toThrow(
```

Comment:
Imho this should not be an error. The expected outcome of the delete operation is, that selected elements won't be on API anymore. If they do not exist and we run delete operation, the outcome is the same which is a valid scenario and not an error.